projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -3075,10 +3081,13 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
         this.deleteRowFromData(rowId, index);
         this._pipeTrigger++;
         this.cdr.markForCheck();
-
+        const dataAfterDelete = !this.transactions.enabled ? data : this.dataWithAddedInTransactionRows;
```

Comment:
The api `get_all_data` method from above already does the same check. Also take `this.dataWithAddedInTransactionRows` only when transactions are not enabled?

Reply:
Also put a comment you need recalculated data with transactions (if any) after the delete.